signalfx/rest.py

```diff
@@ -427,3 +468,124 @@ def delete_detector(self, detector_id, **kwargs):
         resp.raise_for_status()
         # successful delete returns 204, which has no response json
         return resp
+
+    # Functionality related to charts
+    def get_chart(self, chart_id, **kwargs):
+        """Retrieve the single (v2) chart matching the given id.
+
+        Returns:
+            A (v2) chart as a JSON object (usable like a dictionary).
+        """
+        return self._get_object_by_id(self._CHART_ENDPOINT_SUFFIX, chart_id)
+
+    def get_charts(self, name=None, tags=None, batch_size=128, **kwargs):
+        """Retrieve all (v2) charts matching the given name; all (v2) charts up to batch_size otherwise.
+
+        Note that this operation may be expensive.
+        """
+        charts = []
+        offset = 0
+        while True:
+            resp = self._get(
+                self._u(self._CHART_ENDPOINT_SUFFIX),
+                params={
+                    'offset': offset,
+                    'limit': batch_size,
+                    'name': name,
+                    'tags': tags or []
+                },
+                **kwargs)
+            resp.raise_for_status()
+            data = resp.json()
+            charts += data['results']
+            if len(charts) == data['count']:
```

Comment:
This is dangerous if a chart is deleted while this loop pages through charts as `len(charts)` and `data['count']` may never be equal. You should use `>=` instead.